cpu/samd21/periph/pwm.c

```diff
@@ -151,7 +151,7 @@ uint32_t pwm_init(pwm_t dev, pwm_mode_t mode, uint32_t freq, uint16_t res)
     f_real = (CLOCK_CORECLOCK / (scale * res));
 
     /* configure the used pins */
-    for (unsigned i = 0; i < PWM_MAX_CHANNELS; i++) {
+    for (unsigned i = 0; i < pwm_config[dev].chan_numof; i++) {
```

Comment:
Wait, I'm not sure of this change.

Reply:
Why though?

Reply:
I had a doubt on something at first glande but I was wrong. Nevermind.